### PR TITLE
feat: Compile api interface function

### PR DESCRIPTION
**Pull request by oguzbilgener**
Extract the compilation of the API interface to a separate function, in order to make it easier to call Exegesis controllers out of a real Exegesis context.

This commit does not change the behavior of `compileRunner`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -40,6 +40,57 @@ function bundle(
     );
 }
 
+interface CompileApiInterfaceResult {
```
Comment (jwalton) [inline src/index.ts]: If you have a function that returns this interface, you should export the interface as well (and also ExegesisCompiledOptions and PluginsManager).  Or, alternatively, you maybe want to not export this.  :P  Do you need all three of these for what you're doing, or just the apiInterface?

File: src/index.ts [hunk 2]
```diff
@@ -40,6 +40,57 @@ function bundle(
...
+ * @param options - Options.  See docs/options.md
+ * @returns - a Promise<{compiledOptions, apiInterface, plugins}>
+ */
+export function compileApiInterface(
+    openApiDoc: string | oas3.OpenAPIObject,
+    options: ExegesisOptions,
+): Promise<CompileApiInterfaceResult>;
+
+/**
+ * Compiles and API interface for the given openApiDoc using the options.
```
Comment (jwalton) [inline src/index.ts]:
```suggestion
 * Compiles an API interface for the given openApiDoc using the options.
```

File: src/index.ts (hunk 1 above)
Comment (jwalton) [reply src/index.ts]: I'd rather not expose them if I don't have to.  Less surface area for breaking changes.  :)

Comment (oguzbilgener) [reply src/index.ts]: I made it so the new exported function only returns the `apiInterface`. Then I realized I need to export a bunch of other types (involving `ApiInterface`), so you don't have a `types/internal.ts` file anymore 😛

**Pull request merged**